app/controllers/recommendations_controller.rb

```diff
@@ -43,7 +43,12 @@ def edit
   def update
     respond_to do |format|
       @recommendation.assign_attributes(recommendation_params)
-      @recommendation.review!
+
+      if @recommendation.review_complete?
+        @recommendation.review!
+      else
+        @recommendation.save
```

Comment:
What if the recommendation does not save?

It will still execute the below success statement. I think we need to use ! or have some logic to say `if @recommendation.save` etc

Reply:
Oops, you are right. Good shout!